src/System.Net.Security/src/System/Net/Security/SslStreamInternal.cs

```diff
@@ -204,13 +204,16 @@ internal void EndWrite(IAsyncResult asyncResult)
             }
         }
 
-        private void EnsureInternalBufferSize()
+        private void ResetReadBuffer()
         {
+            Debug.Assert(_decryptedBytesCount == 0);
+            Debug.Assert(_internalBuffer == null || _internalBufferCount > 0);
+
             if (_internalBuffer == null)
             {
                 _internalBuffer = ArrayPool<byte>.Shared.Rent(ReadBufferSize);
             }
-            else if (_internalOffset > 0 && _internalBufferCount > 0)
+            else if (_internalOffset > 0)
```

Comment:
Should the `_internalOffset = 0;` below be moved into this `else` branch?

Reply:
Yes now it's only needed there, I have moved it